mdf_matio/validator.py

```diff
@@ -0,0 +1,397 @@
+from datetime import datetime
+import json
+import os
+from tempfile import TemporaryFile
+
+import jsonschema
+
+
+def _remove_nulls(data, skip=None):
+    """Remove all null/None/empty values from a dict or list, except those listed in skip."""
+    if isinstance(data, dict):
+        new_dict = {}
+        for key, val in data.items():
+            new_val = _remove_nulls(val, skip=skip)
+            if new_val is not None or (skip is not None and key in skip):
+                new_dict[key] = new_val
+        return new_dict
+    elif isinstance(data, list):
+        new_list = []
+        for val in data:
+            new_val = _remove_nulls(val, skip=skip)
+            if new_val is not None:
+                new_list.append(new_val)
+        return new_list
+    # Could delete required but empty blocks - services, etc.
+    # elif hasattr(data, "__len__") and len(data) <= 0:
+    #    return None
+    else:
+        return data
+
+
+class Validator:
+    """Validates MDF feedstock.
```

Comment:
I like that you have a "flow" description, please spell it out in more detail.

The class documentation could also use more detail. Please explain it in a level of detail that Tyler or myself could understand what this object does and how to use it.

Reply:
How does this documentation look? I've expanded it, in line with this no longer being an MDF-internal tool.